mp/src/materialsystem/stdshaders/pbr_dx9.cpp

```diff
@@ -0,0 +1,620 @@
+        if (!params[BUMPMAP]->IsDefined())
+            params[BUMPMAP]->SetStringValue("dev/flat_normal");
+
+        // PBR relies heavily on envmaps
+        if (!params[ENVMAP]->IsDefined())
+            params[ENVMAP]->SetStringValue("env_cubemap");
+
+        // Check if the hardware supports flashlight border color
+        if (g_pHardwareConfig->SupportsBorderColor())
+        {
+            params[FLASHLIGHTTEXTURE]->SetStringValue("effects/flashlight_border");
+        }
+        else
+        {
+            params[FLASHLIGHTTEXTURE]->SetStringValue("effects/flashlight001");
+        }
+    };
+
+    // Define shader fallback
+    SHADER_FALLBACK
+    {
+        return 0;
+    };
+
+    SHADER_INIT
+    {
+        PBR_Vars_t info;
+        SetupVars(info);
+
+        Assert(info.flashlightTexture >= 0);
+        LoadTexture(info.flashlightTexture, TEXTUREFLAGS_SRGB);
+
+        Assert(info.bumpMap >= 0);
+        LoadBumpMap(info.bumpMap);
+
+        Assert(info.envMap >= 0);
+        int envMapFlags = g_pHardwareConfig->GetHDRType() == HDR_TYPE_NONE ? TEXTUREFLAGS_SRGB : 0;
+        envMapFlags |= TEXTUREFLAGS_ALL_MIPS;
+        LoadCubeMap(info.envMap, envMapFlags);
+
+        if (info.emissionTexture >= 0 && params[EMISSIONTEXTURE]->IsDefined())
+            LoadTexture(info.emissionTexture, TEXTUREFLAGS_SRGB);
+
+        Assert(info.mraoTexture >= 0);
+        LoadTexture(info.mraoTexture, 0);
+
+        //bool bIsBaseTextureTranslucent = false;
+        if (params[info.baseTexture]->IsDefined())
+        {
+            LoadTexture(info.baseTexture, TEXTUREFLAGS_SRGB);
+
+            //if (params[info.baseTexture]->GetTextureValue()->IsTranslucent())
+            //{
+            //    bIsBaseTextureTranslucent = true;
+            //}
+        }
+
+        if (params[info.specularTexture]->IsDefined())
+        {
+            LoadTexture(info.specularTexture, TEXTUREFLAGS_SRGB);
+        }
+
+        // Set material var2 flags specific to models
+        if (IS_FLAG_SET(MATERIAL_VAR_MODEL))
+        {
+            SET_FLAGS2(MATERIAL_VAR2_SUPPORTS_HW_SKINNING);             // Required for skinning
+            SET_FLAGS2(MATERIAL_VAR2_DIFFUSE_BUMPMAPPED_MODEL);         // Required for dynamic lighting
+            SET_FLAGS2(MATERIAL_VAR2_NEEDS_TANGENT_SPACES);             // Required for dynamic lighting
+            SET_FLAGS2(MATERIAL_VAR2_LIGHTING_VERTEX_LIT);              // Required for dynamic lighting
+            SET_FLAGS2(MATERIAL_VAR2_NEEDS_BAKED_LIGHTING_SNAPSHOTS);   // Required for ambient cube
+            SET_FLAGS2(MATERIAL_VAR2_SUPPORTS_FLASHLIGHT);              // Required for flashlight
+            SET_FLAGS2(MATERIAL_VAR2_USE_FLASHLIGHT);                   // Required for flashlight
+        }
+
+        // Set material var2 flags specific to brushes
+        else
+        {
+            SET_FLAGS2(MATERIAL_VAR2_LIGHTING_LIGHTMAP);                // Required for lightmaps
```

Comment:
Fix this weird spacing, it at first glance didn't look like it was related to the if statement.

Reply:
Fair